rust/src/lib.rs

```diff
@@ -2080,11 +2057,18 @@ impl FfiWalletSummary {
             ptr_from_vec(account_balances)
         };
 
-        let scan_progress = if let Some(progress) = summary.scan_progress() {
-            Box::into_raw(Box::new(FfiScanProgress {
-                numerator: *progress.numerator(),
-                denominator: *progress.denominator(),
-            }))
+        let scan_progress = if let Some(scan_progress) = summary.scan_progress() {
+            if let Some(recovery_progress) = summary.recovery_progress() {
+                Box::into_raw(Box::new(FfiScanProgress {
+                    numerator: *scan_progress.numerator() + *recovery_progress.numerator(),
+                    denominator: *scan_progress.denominator() + *recovery_progress.denominator(),
```

Comment:
This produces strange results if the `scan_progress` and `recovery_progress` denominators are different. I don't know what is best here, but if you want the two progress ratios to be equally weighted, then it would be `numerator: *scan_progress.numerator * *recovery_progress.denominator + *recovery_progress.numerator * *scan_progress.denominator, denominator: 2 * *scan_progress.denominator * *recovery_progress.denominator`.

Reply:
In this particular case, these ratios are unreduced with the denominators being the number of notes in the scan range. So directly adding the numerators and denominators here is fine; this is going to change soon anyway once I rework the types as I suggest in https://github.com/zcash/librustzcash/issues/1559